icepyx/quest/dataset_scripts/argo.py

```diff
@@ -0,0 +1,434 @@
+            # BGC params
+            "bbp470",
+            "bbp470_argoqc",
+            "bbp532",
+            "bbp532_argoqc",
+            "bbp700",
+            "bbp700_argoqc",
+            "bbp700_2",
+            "bbp700_2_argoqc",
+            "bisulfide",
+            "bisulfide_argoqc",
+            "cdom",
+            "cdom_argoqc",
+            "chla",
+            "chla_argoqc",
+            "cndc",
+            "cndc_argoqc",
+            "cndx",
+            "cndx_argoqc",
+            "cp660",
+            "cp660_argoqc",
+            "down_irradiance380",
+            "down_irradiance380_argoqc",
+            "down_irradiance412",
+            "down_irradiance412_argoqc",
+            "down_irradiance442",
+            "down_irradiance442_argoqc",
+            "down_irradiance443",
+            "down_irradiance443_argoqc",
+            "down_irradiance490",
+            "down_irradiance490_argoqc",
+            "down_irradiance555",
+            "down_irradiance555_argoqc",
+            "down_irradiance670",
+            "down_irradiance670_argoqc",
+            "downwelling_par",
+            "downwelling_par_argoqc",
+            "doxy",
+            "doxy_argoqc",
+            "doxy2",
+            "doxy2_argoqc",
+            "doxy3",
+            "doxy3_argoqc",
+            "molar_doxy",
+            "molar_doxy_argoqc",
+            "nitrate",
+            "nitrate_argoqc",
+            "ph_in_situ_total",
+            "ph_in_situ_total_argoqc",
+            "turbidity",
+            "turbidity_argoqc",
+            "up_radiance412",
+            "up_radiance412_argoqc",
+            "up_radiance443",
+            "up_radiance443_argoqc",
+            "up_radiance490",
+            "up_radiance490_argoqc",
+            "up_radiance555",
+            "up_radiance555_argoqc",
+            # all params
+            "all",
+        ]
+        return valid_params
+
+    def _validate_parameters(self, params) -> list:
+        """
+        Checks that the list of user requested parameters are valid.
+
+        Returns
+        -------
+        The list of valid parameters
+        """
+
+        if "all" in params:
+            params = ["all"]
+        else:
+            valid_params = self._valid_params()
+            # checks that params are valid
+            for i in params:
+                assert (
+                    i in valid_params
+                ), "Parameter '{0}' is not valid. Valid parameters are {1}".format(
+                    i, valid_params
+                )
+
+        return params
+
+    # ----------------------------------------------------------------------
+    # Querying and Getting Data
+
+    def search_data(self, params=None, printURL=False) -> str:
+        """
+        Query for available argo profiles given the spatio temporal criteria
+        and other params specific to the dataset.
+
+        Parameters
+        ---------
+        params: list of str, default ["temperature"]
```

Comment:
Suggestion: Add units for temperature and pressure here (I often forget what they are).